lib/statsd.rb

```diff
@@ -125,9 +137,22 @@ def sampled(sample_rate)
 
   def send(stat, delta, type, sample_rate=1)
     sampled(sample_rate) do
-      prefix = "#{@namespace}." unless @namespace.nil?
-      stat = stat.to_s.gsub('::', '.').gsub(RESERVED_CHARS_REGEX, '_')
-      msg = "#{prefix}#{stat}:#{delta}|#{type}#{'|@' << sample_rate.to_s if sample_rate < 1}"
+      stat = stat.to_s.dup
+      stat.gsub!(COLON_COLON_REGEX, ".".freeze)
```

Comment:
We noticed that passing a string to `gsub` actually ends up allocating a Regexp object under the hood anyway, so passing one here should avoid that.